plugins/formal-verification/formver.viper/src/org/jetbrains/kotlin/formver/viper/ast/Predicate.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.viper.ast
+
+import org.jetbrains.kotlin.formver.viper.IntoSilver
+import org.jetbrains.kotlin.formver.viper.MangledName
+import org.jetbrains.kotlin.formver.viper.toScalaOption
+import org.jetbrains.kotlin.formver.viper.toScalaSeq
+
+class Predicate(
```

Comment:
One thing to think about: is this the level of abstraction that is convenient to work with?  I agree this class should exist for the sake of having a 1:1 wrapper around Viper, but perhaps there's something more we should do (at the same level as `ExpEmbedding`)?  Though perhaps `ClassTypeEmbedding` is already that; it probably depends on whether non-class stuff also gets predicates.